Packages/Newtonsoft.Json-for-Unity.Converters/Newtonsoft.Json.UnityConverters.asmdef

```diff
@@ -9,5 +9,22 @@
         "Newtonsoft.Json.dll"
     ],
     "autoReferenced": true,
-    "defineConstraints": []
+    "defineConstraints": [],
+	"versionDefines": [
+        {
+            "name": "com.unity.modules.ai",
+            "expression": "",
+            "define": "AI_MODULE"
+        },
+        {
+            "name": "com.unity.modules.physics",
+            "expression": "",
+            "define": "PHYSICS_MODULE"
+        },
+        {
+            "name": "com.unity.modules.physics2d",
+            "expression": "",
+            "define": "PHYSICS2D_MODULE"
+        }
+    ]
```

Comment:
This is new for me!

It seems to have been introduced in Unity v2019.1.0. As how this is implemented right now, that would require this package to bump the required minimum Unity version from 2018.1 to 2019.1, which leads to backward incompatibility.

Suggest instead to always load in the listed modules when on a lower version than v2019.1, perhaps by using the `UNITY_X_Y_OR_NEWER` define from ["Platform dependent compilation"](https://docs.unity3d.com/Manual/PlatformDependentCompilation.html). On top of this, I would really prefer a comment above each `#if` statement, something like: "Relies on 'versionDefines' from package.json, which was added in Unity 2019.1.

nitpick: The official Newtonsoft.Json repo, and my fork Newtonsoft.Json-for-Unity, [uses the `HAVE_` prefix](https://github.com/jilleJr/Newtonsoft.Json-for-Unity/blob/7f7f6490cd83b0f1206ca111050795e2a34266c8/Src/Newtonsoft.Json/Newtonsoft.Json.csproj#L52). So I suggest using that here as well.

Also, suggest to do kind of like "reverse domain name" format, where the most volatile part of the string comes last. I.e. changing `_MODULE` suffix to a `MODULE_` prefix.

All in all, I think you should change this to:

```cs
#if HAVE_MODULE_PHYSICS || !UNITY_2019_1_OR_NEWER
#if HAVE_MODULE_PHYSICS2D || !UNITY_2019_1_OR_NEWER
#if HAVE_MODULE_AI || !UNITY_2019_1_OR_NEWER
```